blog/2022/storage-map.adoc

```diff
@@ -0,0 +1,127 @@
+:title: New storage in Keycloak
+:date: 2022-07-27
+:publish: true
+:author: Hynek Mlnařík
+
+Store in Keycloak has known limitations:
+For small deployments, it takes too long to initialize the database and
+start Keycloak. There is no native support for cloud-native deployment.
+Upgrading to a new Keycloak version often means requirement for stop-the-world
+updates with little chance for no downtime upon upgrade.
+
+Keycloak 19 brings in an early preview of the future Keycloak store supporting
+no-downtime upgrades, per-realm storage, and cloud-nativity from its very inception.
+This so caled _map_ storage stays focused at delivering optimal experience and thus
+limits its support to Postgres and CockroachDB databases and Infinispan datastore.
+The early preview is available in Quarkus distribution.
+
+This early preview lacks several features necessary for optimal performance
+and is thus not yet production ready; still we have a plan to deliver those:
+
+- In Keycloak 20, expect improved support especially for CockroachDB;
+  and also a file-based store. Together with another anticipated feature
+  - _tree_ store - it would be possible to combine several storage mechanisms
+  and have e.g. several static client declarations in static files managed
```

Comment:
I remember that Andrew was always picky to avoid Latin abreviations.

```suggestion
  and have, for example, several static client declarations in static files managed
```

Reply:
You are correct, yet I will stick with those in the blog post